### DISCUSSION TITLE
Is there any way to run NUnit?

### DISCUSSION THREAD

**Discussion by DGwynnJones** [community/community#25059] [Actions]
<p>I have a large number of .Net Framework apps with NUnit tests.</p>
<p>I have a test solution with tests compiling – is it possible to run the tests?</p>
<p>I guess I could inlude the packages forder with the console runner but it is there a neater way??</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/dgwynnjones">@dgwynnjones</a> ,</p>
<p>You can get the full path of test project files and execute test then.</p>
<p>Code sample as below, more details <a href="https://github.com/Microsoft/vstest/issues/705#issuecomment-369230120" rel="nofollow noopener">here</a>. It works on my side, hope it helps!</p>
<pre><code>Get-ChildItem "LegacyProjTests" -recurse | where {$_.extension -eq ".csproj"} | % { dotnet test $_.FullName }
</code></pre>
<p>Thanks.</p>
